src/codegen/mod.rs

```diff
@@ -866,8 +866,10 @@ impl CodeGenerator for CompInfo {
                 let item = quote_item!(ctx.ext_cx(),
                                        #[test]
                                        fn $fn_name() {
-                                           assert_eq!($size_of_expr, $size);
-                                           assert_eq!($align_of_expr, $align);
+                                           assert_eq!($size_of_expr, $size,
+                                                      concat!("Size of template specialization: ", stringify!($ident)));
+                                           assert_eq!($align_of_expr, $align,
+                                                      concat!("Alignment of template specialization: ", stringify!($ident)));
                                        })
```

Comment:
The patch is good to me, something else observed. 

Do we need check the field offset for the template specialization?

It seems be different to VTable case, should have potential layout issue. Anyway, I haven't user cases in wild